pkg/server/backend_manager.go

```diff
@@ -121,17 +120,39 @@ func (b *backend) Context() context.Context {
 	return b.conn.Context()
 }
 
-func newBackend(conn agent.AgentService_ConnectServer) *backend {
+func (b *backend) GetAgentIdentifiers() (header.Identifiers, error) {
+	var agentIdentifiers header.Identifiers
+	md, ok := metadata.FromIncomingContext(b.Context())
+	if !ok {
+		return agentIdentifiers, fmt.Errorf("failed to get context")
+	}
+	agentIDs := md.Get(header.AgentIdentifiers)
+	if len(agentIDs) > 1 {
+		return agentIdentifiers, fmt.Errorf("expected at most one agent IP in the context, got %v", agentIDs)
```

Comment:
```suggestion
		return agentIdentifiers, fmt.Errorf("expected at most one agent ID in the context, got %v", agentIDs)
```

Actually... this is expecting 1 list of agent IDs, so maybe this is more accurate:

```suggestion
		return agentIdentifiers, fmt.Errorf("expected at most one set of agent IDs in the context, got %v", agentIDs)
```

Alternatively, we could just allow this:

```suggestion
		agentIDs = []string{strings.Join(agentIDs, ",")}
```

Reply:
Done (except the last suggestion - just to reduce risk).